### PR TITLE
Signer new-token generates a link and opens browser

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes https://github.com/ethcore/parity/issues/2874

### CODE REVIEW

File: parity/signer.rs [hunk 1]
```diff
@@ -71,12 +72,24 @@ fn codes_path(path: String) -> PathBuf {
...
+	let code = try!(generate_new_token(cmd.path).map_err(|err| format!("Error generating token: {:?}", err)));
+	let auth_url = format!("http://{}:{}/#/auth?token={}", cmd.signer_interface, cmd.signer_port, code);
+	// Open a browser
+	url::open(&auth_url);
+	// And print in to the console
+	Ok(format!(
+		r#"
+Open: {}
+to authorize your browser.
+Or use the code: {}"#,
```
Comment (rphmeier) [inline parity/signer.rs]: Have also had a few requests to put the code on its own line to make it easier to copy

**Comment by tomusdrw** [comment]
So I think we should rather:
1. Remove `parity signer new-token`
2. Have `parity ui` open UI with newly generated token each time
3. But we also need to keep track of token creation date and number of times it was used
4. Remove old tokens that have never been used.

**Comment by jacogr** [comment]
UI code to handle the auth token as passed forthcoming in this PR.

**Comment by jacogr** [comment]
Closes https://github.com/ethcore/parity/issues/3400

**Comment by tomusdrw** [comment]
1. `parity ui` generates new token and opens up a browser with UI (in case parity is running it just opens a browser)
2. `parity signer new-token` generates token, opens browser and quits.
3. We track creation time and usage of tokens.
4. Tokens which were not used and are more then a day old are removed.

**Comment by gavofyork** [comment]
`parity signer new-token` generates token, opens browser and quits

so a user *doesn't care* about authorisation tokens - they only care about using parity's UI. if this opens a browser window, it should be called `parity open` or some such. keeping it with the present name:
a) confuses any power-users that expect it just to create a new token; but _way_ more importantly:
b) is a really obscure command for actually opening a new window.

**Comment by tomusdrw** [comment]
`parity ui` - opens a browser with new token in URL (starts parity if not running already)
`parity signer new-token` - displays the new token (and link for convenience) and quits

**Pull request merged**